src/generator.cs

```diff
@@ -4353,6 +4356,7 @@ void GenerateProperty (Type type, PropertyInfo pi, List<string> instance_fields_
 				PrintPlatformAttributes (pi);
 				PrintPlatformAttributes (pi.GetGetMethod ());
 				PrintPreserveAttribute (pi.GetGetMethod ());
+				PrintAdviceAttribute (pi.GetGetMethod ());
```

Comment:
That's a 3rd `pi.GetGetMethod ()` in a row, not sure if it's cached or not (even less sure when we switch to IKVM), but it's hard to see any penalty in calling it only once.

Also I wonder if we should not have a `PrintAttribute` so adding the next _global_ one can be done in a single (or fewer) place(s).